### ISSUE TITLE
Unreachability should detect redundant `isinstance`s even for `object` arguments

### ISSUE THREAD

**Issue by A5rocks** [python/mypy#20668]
**Bug Report**

See #20665 

**To Reproduce**

```python
def f(x: object) -> None:
    if isinstance(x, int):
        print("aaaa")
    elif isinstance(x, int):
        print("aaaa")
```

**Expected Behavior**

Error

**Actual Behavior**

No error

**Your Environment**

- Mypy version used: master
- Mypy command-line flags: `--warn-unreachable --strict`
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.13

**Comment by A5rocks** [comment]
Ugh, nevermind. Properly solving this requires negation types.
